### ISSUE TITLE
Should try/except only catch ImportError?

### ISSUE THREAD

**Issue by eronsdc** [shap/shap#358]
I was getting the error

`XGBoost could not be imported!`

due to this [code block](https://github.com/slundberg/shap/blob/master/shap/explainers/tree.py#L16) in explainers.tree.py :

```
try:
    import xgboost
except Exception as e:
    import_errors["xgboost"] = "XGBoost could not be imported!", e
```

I can in fact import xgboost, but there is currently a DeprecationWarning coming via sklearn dependency somewhere for my current enviroment:
`C:\Users\XXX\AppData\Local\Continuum\anaconda3\envs\ds\lib\site-packages\sklearn\externals\joblib\externals\cloudpickle\cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
`

I get the feeling that the `Exception` was not intended to catch such warning exceptions. Would it be a better practice to replace the `Exception` with `ImportError`?

I am no pyhton expert, so any feedback would be appreciated.

**Comment by slundberg** [comment]
Great point. When I wrote that I didn't think about warnings being a subtype of `Exception`.

**Comment by eronsdc** [comment]
I wouldn't mind having a go at changing these if you like. Its probably not that important at the end of the day, but I don't mind doing a quick little PR for it.

**Comment by slundberg** [comment]
@eronsdc Thanks! I actually already fixed it in 7da342e so we should be good once the next release is pushed :)
